src/metpy/calc/indices.py

```diff
@@ -126,20 +126,17 @@ def mean_pressure_weighted(pressure, *args, height=None, bottom=None, depth=None
        Renamed ``heights`` parameter to ``height``
 
     """
-    ret = []  # Returned variable means in layer
     layer_arg = get_layer(pressure, *args, height=height,
                           bottom=bottom, depth=depth)
-    layer_p = layer_arg[0]
-    layer_arg = layer_arg[1:]
+    layer_p = layer_arg[0]  # Pressure profile
+    layer_arg = layer_arg[1:]  # All other profiles provided to args
+
```

Comment:
How about using Python 3 unpacking here (and combine with the call to `get_layer`:
```suggestion
    # Split pressure profile from other variables to average
    layer_p, *others = get_layer(pressure, *args, height=height, bottom=bottom, depth=depth)
```

Would have to update my other suggestion with `others`.